src/pykeen/nn/functional.py

```diff
@@ -1294,3 +1295,39 @@ def cp_interaction(
         r,
         t,
     )
+
+
+def triple_re_interaction(
+    # head
+    h: torch.FloatTensor,
+    # relation
+    r_head: torch.FloatTensor,
+    r_mid: torch.FloatTensor,
+    r_tail: torch.FloatTensor,
+    # tail
+    t: torch.FloatTensor,
+    # version 2: relation factor offset
+    u: Optional[float] = None,
+    # extension: negative (power) norm
+    p: int = 2,
+    power_norm: bool = False,
+) -> torch.FloatTensor:
+    # cf. https://github.com/LongYu-360/TripleRE-Add-NodePiece/blob/994216dcb1d718318384368dd0135477f852c6a4/TripleRE%2BNodepiece/ogb_wikikg2/model.py#L317-L328  # noqa: E501
+    # normalize head & tail
+    h = functional.normalize(h, dim=-1, p=2)
```

Comment:
is this a property of the interaction or the representation?

Reply:
The original publication makes this part of the interaction, but I agree that we should move that to the model